pkg/controller/workloaddispatcher/incrementaldispatcher_test.go

```diff
@@ -0,0 +1,289 @@
+			mkAcState: &kueue.AdmissionCheckState{
+				Name:  "ac1",
+				State: kueue.CheckStatePending,
+			},
+			remoteClusters: []string{"cluster1"},
+			clusters: []kueue.MultiKueueCluster{
+				*utiltesting.MakeMultiKueueCluster("cluster1").
+					KubeConfig(kueue.SecretLocationType, "cluster1").
+					Generation(1).
+					Obj(),
+			},
+		},
+		{
+			name:           "workload has quota reserved",
+			dispatcherName: kueueconfig.MultiKueueDispatcherModeIncremental,
+			workload:       baseWorkloadBuilder.DeepCopy(),
+			mkAcState: &kueue.AdmissionCheckState{
+				Name:  "ac1",
+				State: kueue.CheckStatePending,
+			},
+			remoteClusters: []string{"cluster1"},
+			clusters: []kueue.MultiKueueCluster{
+				*utiltesting.MakeMultiKueueCluster("cluster1").
+					KubeConfig(kueue.SecretLocationType, "cluster1").
+					Generation(1).
+					Obj(),
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			objs := []runtime.Object{}
+			if tc.mkAcState != nil {
+				tc.workload.Status.AdmissionChecks = []kueue.AdmissionCheckState{*tc.mkAcState}
+				ac := utiltesting.MakeAdmissionCheck(string(tc.mkAcState.Name)).
+					ControllerName(kueue.MultiKueueControllerName).
+					Parameters(kueue.GroupVersion.Group, "MultiKueueConfig", string(tc.mkAcState.Name)).
+					Obj()
+
+				objs = append(objs, ac)
+			}
+
+			if tc.workload != nil {
+				objs = append(objs, tc.workload)
+			}
+
+			if tc.mkAcState != nil {
+				mkConfig := utiltesting.MakeMultiKueueConfig(string(tc.mkAcState.Name)).Clusters("cluster1").Obj()
+				objs = append(objs, mkConfig)
+			}
+			scheme := runtime.NewScheme()
+			utilruntime.Must(kueue.AddToScheme(scheme))
+
+			if tc.clusters != nil {
+				for _, cluster := range tc.clusters {
+					objs = append(objs, &cluster)
+				}
+			}
+			cl := fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(objs...).Build()
+			helper, _ := admissioncheck.NewMultiKueueStoreHelper(cl)
+			rec := &IncrementalDispatcherReconciler{
+				client:          cl,
+				helper:          helper,
+				clock:           fakeClock,
+				dispatcherName:  tc.dispatcherName,
+				roundStartTimes: make(map[types.NamespacedName]time.Time),
+			}
+
+			req := ctrl.Request{NamespacedName: types.NamespacedName{Name: workloadName, Namespace: testNamespace}}
+			ctx, _ := utiltesting.ContextWithLog(t)
+			_, gotErr := rec.Reconcile(ctx, req)
+			if diff := cmp.Diff(tc.wantErr, gotErr); diff != "" {
+				t.Errorf("Unexpected error (-want/+got)\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestIncrementalDispatcherNominateWorkers(t *testing.T) {
+	const (
+		testNamespace = "default"
+		testName      = "test-wl"
+	)
+
+	testCases := []struct {
+		name               string
+		remoteCount        int
+		preNominated       []string
+		expectNominated    int
+		wantErr            error
+		advanceRoundTime   bool
+		expectNominatedSet []string
+	}{
+		{"one remote", 1, nil, 1, nil, false, []string{"A"}},
```

Comment:
```suggestion
		"one remote": {
			remoteCount: 1,
			previousRoundNominatedClusters: nil,
			wantNominatedClustersCount: 1,
			wantErr: nil,
			advanceRountTime: false,
			wantNominatedClusters: []string{"A"},
		},
```

Could you specify typed for debuggability?